addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -3155,20 +3155,36 @@ registry.SnippetSave = SnippetOptionWidget.extend({
                     classes: 'btn-primary',
                     close: true,
                     click: async () => {
-                        const snippetName = dialog.el.querySelector('.o_we_snippet_name_input').value;
-                        const targetCopyEl = this.$target[0].cloneNode(true);
-                        delete targetCopyEl.dataset.name;
-                        await this._rpc({
-                            model: 'ir.ui.view',
-                            method: 'save_snippet',
-                            kwargs: {
-                                'name': snippetName,
-                                'arch': targetCopyEl.outerHTML,
-                                'template_key': this.options.snippets,
-                                'snippet_class': [...this.$target[0].classList].filter(x => /\bs_./g.test(x))[0],
+                        const save = await this._confirmReload();
+                        if (!save) {
+                            return;
+                        }
+                        // Hack: by the time onSuccess is called after request_save, the
+                        // current widget has been destroyed and is orphaned, so this._rpc
+                        // will not work as it can't trigger_up. To solve this we get a
+                        // reference to the rpc service provider before requesting the save
+                        let serviceProvider = this.getParent();
+                        while (!serviceProvider._call_service) {
+                            serviceProvider = serviceProvider.getParent();
+                        }
+                        this.trigger_up('request_save', {
+                            reloadEditor: true,
+                            onSuccess: async () => {
```

Comment:
Maybe `await` the success of this request then do the snippet save? To avoid touching (indenting) the snippet saving code ;)

Reply:
Unfortunately, I need this option method to return for the request_save to be able to do its job, because save_request waits for the editor's mutex to unlock. Awaiting save_request in an option method results in a deadlock on the mutex.

Reply:
Mmh, too bad then.